### ISSUE TITLE
[MC-248655] Minecarts behave inconsistently whilst changing directions

### ISSUE THREAD

**Issue by Daniel99j2** [fxmorin/carpet-fixes#174]
**Does your bug have a Mojira id? If so, what is it?**
It's [MC-248655](https://bugs.mojang.com/browse/MC-248655)

**What's the bug about? Why do you want it fixed?**
When minecarts turn, they can trigger detector rails they are past (and probably other issues, but I don't know them)
If you try to turn stacked chest minecarts, they fly off (unrelated issue), so you can make a contraption in the screenshot.
It works on some directions, but not in others.

**Describe the solution you'd like**
When minecarts turn, they are the same in all directions

**Describe alternatives you've considered**
There are not many ways to turn stacked chest minecarts

**Additional context**
Related to https://bugs.mojang.com/browse/MC-258973
You can see it clearly in https://www.mediafire.com/file/glre0lqagtsvppo/piston_bolt_demo_world.zip/file
at /execute in minecraft:overworld run tp @s 32.24 15.00 26.33 -428.95 55.20
then create arrangement like:
![2024-04-26_06 57 28](https://github.com/FxMorin/carpet-fixes/assets/121533735/b4dd55d8-3786-4c32-a3f3-0277a0051bef)
on the south-west corner

**What should the rule be called?**
minecartsTurnConsistentlyFix

**Comment by FxMorin** [comment]
Yes this is an interesting one, I don't have the time to work on it currently.

**Comment by FxMorin** [comment]
Oh wait I might have already fixed this. Try `detectorRailDetectsTooEarlyFix`

**Comment by Daniel99j2** [comment]
Sadly that rule is does not fix the bug

**Comment by FxMorin** [comment]
Mojang fixed it before I could
